src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -796,7 +796,7 @@ function BaseSelectionList<TItem extends ListItem>(
         shouldBubble: !flattenedSections.allOptions.at(focusedIndex) || focusedIndex === -1,
         shouldStopPropagation,
         shouldPreventDefault,
-        isActive: !disableKeyboardShortcuts && !disableEnterShortcut && isFocused,
+        isActive: !disableKeyboardShortcuts && !disableEnterShortcut && isFocused && focusedIndex >= 0,
```

Comment:
The root cause of why the `onSubmitEditing` isn't called is because we have an ENTER keyboard shortcut on the selection list that prevents default the Enter key, even though there is no item focused. After fixing this, when pressing Enter,

1. If there is a focused item, then `onSubmitEditing` won't be called.
2. Else, `onSubmitEditing` will be called.

So, only 1 `submitSearch` will be triggered.

Reply:
hey! a bit late but that was a nice RCA and a nice fix 👍 I remember the ENTER behavior on search input was wonky for a long time 😅